File: GSN.py
from colorama import init, deinit, Fore
from os import system, name
from random import choice, randint

__VERSION__ = '1.0.1'
__AUTHOR__ = 'workonfire'


def color_print(color, text):
    init(autoreset=True)
    print(color + text)
    deinit()


def main():
    if name == 'nt':
        system('title GSN')

    color_print(Fore.WHITE, "         (        )  ")
    color_print(Fore.LIGHTYELLOW_EX, " (       )\\ )  ( /(  ")
    color_print(Fore.YELLOW, " )\\ )   (()/(  )\\()) ")
    color_print(Fore.LIGHTRED_EX, "(()/(    /(_))((_)\\  ")
    color_print(Fore.RED, " /(_))_ (_))   _((_) ")
    color_print(Fore.RED, "(_)) __|/ __| | \\| | ")
    color_print(Fore.RED, "  | (_ |\\__ \\ | .` | ")
    color_print(Fore.RED, "   \\___||___/ |_|\\_| ")
    print("\nGenerator Spierdolonych Nicków v{}".format(__VERSION__))
    color_print(Fore.LIGHTYELLOW_EX, "by {}\n".format(__AUTHOR__))

    while True:
        try:
            how_many_times = int(input("Ilość nicków do wygenerowania: "))
            max_length = int(input("Maksymalna ilość znaków: "))
            color_print(Fore.RED,
                        "UWAGA: Jeśli jakieś nicki będą przekraczały ustalony limit znaków, zostaną one przycięte.")
            save_to_file = input("Zapisać wynik do pliku? (y/n) ").lower().rstrip()
            generate_number = input("Doklejać losową liczbę? (y/n) ").lower().rstrip()
            break
        except ValueError:
            color_print(Fore.RED, "Podaj poprawne wartości.")

    with open("dictionaries/adjectives.txt") as adjectives_file:
        adjectives = [line.rstrip() for line in adjectives_file.readlines()]
    with open("dictionaries/nouns.txt") as nouns_file:
        nouns = [line.rstrip() for line in nouns_file.readlines()]

    generated_nicknames = []
    print("Generowanie nicków...\n")

    for i in range(how_many_times):
        adjective = choice(adjectives)
        noun = choice(nouns)
        noun = '_' + noun if choice(['underline', 'capitalize']) == 'underline' else noun.title()
        generated_nick = adjective + noun
        if generate_number == 'y':
            generated_nick += str(randint(1, 100))
        if len(generated_nick) > max_length:
            generated_nick = generated_nick[:max_length]
        generated_nicknames.append(generated_nick)

    for nick in generated_nicknames:
        color_print(Fore.LIGHTCYAN_EX, "> " + nick)

    color_print(Fore.LIGHTGREEN_EX, "\nGotowe.")

    if save_to_file == 'y':
        generated_nicknames = "\n".join(generated_nicknames)
        with open("output.txt", 'a') as log_file:
            log_file.write(generated_nicknames)
        color_print(Fore.LIGHTGREEN_EX, "Zapisano nicki do pliku.")

    if name == 'nt':
        print("By zakończyć pracę programu, naciśnij dowolny klawisz.")
        system('pause >nul')


if __name__ == '__main__':
    try:
        main()
    except KeyboardInterrupt:
        pass

File: kotlin/src/pl/workonfire/gsn/GSN.kt
package pl.workonfire.gsn

import org.apache.commons.lang3.StringUtils
import org.fusesource.jansi.Ansi
import org.fusesource.jansi.Ansi.Color.*
import org.fusesource.jansi.AnsiConsole
import java.io.File
import java.io.IOException
import java.util.Random
import kotlin.collections.ArrayList
import kotlin.system.exitProcess

object GSN {
    private const val version = "1.0.2"
    private const val author = "workonfire"

    private fun String.colorPrint(bright: Boolean, color: Ansi.Color?) {
        println(
            if (bright)
                Ansi.ansi().fgBright(color).a(this).reset()
            else
                Ansi.ansi().fg(color).a(this).reset()
        )
    }

    fun run() {
        val howManyTimes: Int
        val maxLength: Int
        val saveToFile: Boolean
        val appendNumber: Boolean
        val stripPolishChars: Boolean
        val adjectives: List<String>
        val nouns: List<String>

        AnsiConsole.systemInstall()

        "         (        )  ".colorPrint(false, WHITE)
        " (       )\\ )  ( /(  ".colorPrint(true, YELLOW)
        " )\\ )   (()/(  )\\()) ".colorPrint(false, YELLOW)
        "(()/(    /(_))((_)\\  ".colorPrint(true, RED)
        " /(_))_ (_))   _((_) ".colorPrint(false, RED)
        "(_)) __|/ __| | \\| | ".colorPrint(false, RED)
        "  | (_ |\\__ \\ | .` | ".colorPrint(false, RED)
        "   \\___||___/ |_|\\_| \n".colorPrint(false, RED)

        println("Generator Spierdolonych Nicków v$version")
        "by $author\n".colorPrint(true, YELLOW)

        while (true) {
            print("Ilość nicków do wygenerowania: ")
            howManyTimes = readLine()?.toInt() ?: 0
            if (howManyTimes == 69) {
                "▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄".colorPrint(true, RED)
                "██ ███ █▄▄ ██▄██ ██ █ ▄▀▄ ██".colorPrint(true, YELLOW)
                "██ █ █ █▀▄███ ▄█ ██ █ █▄█ ██".colorPrint(true, GREEN)
                "██▄▀▄▀▄█▄▄▄█▄▄▄██▄▄▄█▄███▄██".colorPrint(true, BLUE)
                "▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀".colorPrint(true, MAGENTA)
            }
            print("Maksymalna ilość znaków: ")
            maxLength = readLine()?.toInt() ?: 0
            "UWAGA: Jeśli jakieś nicki będą przekraczały ustalony limit znaków, zostaną one przycięte."
                .colorPrint(false, RED)
            print("Zapisać wynik do pliku? (y/n) ")
            saveToFile = readLine()?.toLowerCase() == "y"
            print("Doklejać losową liczbę? (y/n) ")
            appendNumber = readLine()?.toLowerCase() == "y"
            print("Usunąć polskie znaki? (y/n) ")
            stripPolishChars = readLine()?.toLowerCase() == "y"
            break
        }
        try {
            adjectives = File("dictionaries/adjectives.txt").readLines(charset = Charsets.UTF_8)
            nouns = File("dictionaries/nouns.txt").readLines(charset = Charsets.UTF_8)
        } catch (exception: IOException) {
            "Wystąpił problem podczas próby otworzenia słowników.".colorPrint(false, RED)
            exitProcess(1)
        }
        val generatedNicknames: MutableList<String> = ArrayList()
        println("Generowanie nicków...\n")
        for (i in 1 until howManyTimes) {
            val randomAdjectiveIndex = Random().nextInt(adjectives.size)
            val adjective = adjectives[randomAdjectiveIndex]
            val randomNounIndex = Random().nextInt(nouns.size)
            var noun = nouns[randomNounIndex]
            noun = if (Random().nextBoolean())
                "_$noun"
            else
                noun.substring(0, 1).toUpperCase() + noun.substring(1)
            var generatedNick = adjective + noun
            if (appendNumber) generatedNick += Random().nextInt(100) + 1
            if (generatedNick.length > maxLength) generatedNick = generatedNick.substring(0, maxLength)
            if (stripPolishChars) generatedNick = StringUtils.stripAccents(generatedNick)
            generatedNicknames.add(generatedNick.replace("\\s+".toRegex(), ""))
        }
        for (nick in generatedNicknames) ">> $nick".colorPrint(true, CYAN)
        "\nGotowe.".colorPrint(false, GREEN)
        if (saveToFile) {
            try {
                File("output.txt").printWriter().use { out ->
                    generatedNicknames.forEach {
                        out.println(it)
                    }
                }
                "Zapisano nicki do pliku.".colorPrint(false, GREEN)
            } catch (exception: IOException) {
                "Wystąpił problem podczas zapisywania pliku.".colorPrint(false, RED)
            }
        }
        println("By zakończyć pracę programu, naciśnij Enter.")
        readLine()
        AnsiConsole.systemUninstall()
    }
}

fun main() {
    GSN.run()
}